staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -1815,8 +1815,14 @@ func TestForgetWatcher(t *testing.T) {
 	require.NoError(t, err)
 	defer cacher.Stop()
 
-	require.Equal(t, 0, len(cacher.watchers.allWatchers))
-	require.Equal(t, 0, len(cacher.watchers.valueWatchers))
+	assertCacherInternalState := func(expectedWatchersCounter, expectedValueWatchersCounter int) {
```

Comment:
it is a low level test so we need to acquire the lock before messing with the internal state.